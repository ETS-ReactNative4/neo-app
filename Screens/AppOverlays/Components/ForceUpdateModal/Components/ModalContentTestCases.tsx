import React from "react";
import { ITestCase } from "../../../../../TypeInterfaces/TestCases/ITestCases";
import { StyleSheet, Alert } from "react-native";

import { CONSTANT_shade4 } from "../../../../../constants/colorPallete";
import ModalContent from "./ModalContent";

const styles = StyleSheet.create({
  modalContainerStyle: {
    backgroundColor: CONSTANT_shade4
  }
});

const ModalContentTestCases: ITestCase[] = [
  {
    title: "Force update modal content",
    Component: (
      <ModalContent
        containerStyle={styles.modalContainerStyle}
        title={"Your app is outdated"}
        description={
          "You are using an old version of the app. We have added new features to enhance the app experience."
        }
        buttonText={"Update now"}
        buttonClickAction={() => Alert.alert("Click Button")}
        bottomText={"Having probelm in updating the app?"}
        linkText={"Message us"}
        linkClickAction={() => Alert.alert("Click Link")}
      />
    )
  }
];

export default ModalContentTestCases;
