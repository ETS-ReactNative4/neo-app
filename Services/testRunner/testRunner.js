import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
