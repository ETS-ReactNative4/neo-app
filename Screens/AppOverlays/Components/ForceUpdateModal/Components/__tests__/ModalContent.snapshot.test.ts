import ModalContentTestCases from "../ModalContentTestCases";
import testRunner from "../../../../../../Services/testRunner/testRunner";

ModalContentTestCases.map(testRunner);
