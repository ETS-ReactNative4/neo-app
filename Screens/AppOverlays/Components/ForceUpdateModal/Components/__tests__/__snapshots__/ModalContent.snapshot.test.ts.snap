// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Force update modal content 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(247, 247, 247)",
        "borderRadius": 8,
        "flexDirection": "column",
        "paddingHorizontal": 32,
        "paddingVertical": 24,
      },
      Object {
        "backgroundColor": "rgba(221, 221, 221, 1)",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "rgba(44, 47, 66, 1)",
        "fontFamily": "SourceSansPro-SemiBold",
        "fontSize": 16,
        "lineHeight": 16,
        "marginBottom": 16,
      }
    }
  >
    Your app is outdated
  </Text>
  <Text
    ellipsizeMode="tail"
    numberOfLines={3}
    style={
      Object {
        "color": "rgba(83, 87, 109, 1)",
        "fontFamily": "SourceSansPro-Regular",
        "fontSize": 15,
        "lineHeight": 20,
        "marginBottom": 16,
        "textAlign": "center",
      }
    }
  >
    You are using an old version of the app. We have added new features to enhance the app experience.
  </Text>
  <View>
    <TouchableOpacity
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "height": 56,
          },
          Object {
            "height": 44,
            "marginBottom": 16,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 56,
            },
            Object {
              "backgroundColor": "rgba(0,198,132,1)",
              "borderRadius": 4,
              "justifyContent": "center",
              "paddingHorizontal": 24,
            },
            Object {
              "height": 44,
              "marginBottom": 16,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "rgb(247, 247, 247)",
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 17,
                "lineHeight": 17,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 15,
                "lineHeight": 15,
              },
            ]
          }
        >
          Update now
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;
